components/TextFields/src/MDCTextInput.h

```diff
@@ -0,0 +1,138 @@
+
+/**
+ The Material Design guidelines have many suggestions for handling text input. The inputs that
+ conform to this protocol have all the API necessary to achieve those suggestions.
+
+ They are, however, dumb; they do not handle error state nor validation.
+
+ - For handling error states and other Material behaviors use an MDCTextInputController on your
+ inputs.
+ - For validation, there are many 3rd party libraries you can use like:
+   - https://github.com/3lvis/Validation
+   - https://github.com/adamwaite/Validator
+ */
+
+@class MDCTextInputUnderlineView;
+
+@protocol MDCTextInputPositioningDelegate;
+
+/** Common API for Material Design compliant text inputs. */
+@protocol MDCTextInput <NSObject>
+
+/**
+ The attributed text string of the placeholder label.
+ Bringing convenience api found in UITextField to all MDCTextInputs. Maps to the .attributedText of
+ the
+ placeholder label.
+ */
+@property(nonatomic, nullable, copy) NSAttributedString *attributedPlaceholder;
+
+/** The text displayed in the text input with style attributes. */
+@property(nonatomic, nullable, copy) NSAttributedString *attributedText;
+
+/**
+ A button that can appear inline that when touched clears all entered text and resets the input to
+ an empty state.
+ */
+@property(nonatomic, nonnull, strong, readonly) UIButton *clearButton;
+
+/**
+ Color for the "clear the text" button image.
+
+ Color changes are not animated.
+
+ Default is black with 38% opacity.
+ */
+@property(nonatomic, nullable, strong) UIColor *clearButtonColor UI_APPEARANCE_SELECTOR;
+
+/**
+ Controls when the clear button will display.
+
+ Note: The clear button will never display when there is no entered text.
+ */
+@property(nonatomic, assign) UITextFieldViewMode clearButtonMode;
+
+/** A Boolean value indicating whether the text field is currently in edit mode. */
+@property(nonatomic, assign, readonly, getter=isEditing) BOOL editing;
+
+/** The font of the text in the input. */
+@property(nonatomic, nullable, strong) UIFont *font;
+
+/** Should it have the standard behavior of disappearing when you type? Defaults to YES. */
+@property(nonatomic, assign) BOOL hidesPlaceholderOnInput;
+
+/**
+ The label on the leading side under the input.
+
+ This will usually be used for placeholder text to be displayed when no text has been entered. The
+ Material Design guidelines call this 'helper text.'
+ */
+@property(nonatomic, nonnull, strong, readonly)
+    UILabel *leadingUnderlineLabel NS_SWIFT_NAME(leadingLabel);
+
+/*
+ Indicates whether the alert contents should automatically update their font when the device’s
+ UIContentSizeCategory changes.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ Default value is NO.
+ */
+@property(nonatomic, setter=mdc_setAdjustsFontForContentSizeCategory:)
+    BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;
```

Comment:
UI_APPEARANCE_SELECTOR or IBInspectable?  
I can imagine a instance where some interfaces support Dynamic Text while others do not.
Here and below.